File: server/web/models/photo.js
const { Model, DataTypes } = require('sequelize');
const sequelize = require('../config/connection');

class Photo extends Model {}

Photo.init(
  {
    id_photo: {
      type: DataTypes.INTEGER,
      allowNull: false,
      primaryKey: true,
      autoIncrement: true,
    },
    url: {
      type: DataTypes.STRING,
    },
    id_post: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: 'Post',
        key: 'id_post',
      },
    },
  },
  {
    sequelize,
    timestamps: true,
    freezeTableName: true,
    underscored: true,
    modelName: 'Photo',
  }
);

module.exports = Photo;

File: social_network_app/lib/data/models/post/photo.dart
class Photo {
  int? id_photo;
  String? url;
  String? belong;
  int? id_post;

  Photo({this.id_photo, this.url, this.belong, this.id_post});

  factory Photo.fromJson(Map<String, dynamic> json) {
    return Photo(
        url: json["url"] ?? "",
        belong: json["belong"] ?? "",
        id_post: json["id_post"] ?? 0,
        id_photo: json["id_photo"] ?? 0);
  }
}